src/main/kotlin/org/rust/ide/intentions/AddDerive.kt

```diff
@@ -0,0 +1,62 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.CodeStyleManager
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.appendMetaToList
+import org.rust.lang.core.psi.util.parentOfType
+
+class AddDerive : PsiElementBaseIntentionAction()
+                , IntentionAction {
+    override fun getFamilyName() = "Add derive clause"
+    override fun getText() = "Add Derive Clause"
+    override fun startInWriteAction() = true
+
+    override fun invoke(project: Project, editor: Editor?, element: PsiElement) {
+        val item = element.parentOfType<RustItem>() ?: return
+        val existingDeriveAttr = item.outerAttrList.find { it.metaItem?.identifier?.textMatches("derive") ?: false }
```

Comment:
There is an API for querying item attributes: https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/psi/impl/mixin/RustItemImplMixin.kt#L35-L53.

It could be extended to have a function like `findAttribute(name: String) : RustMetaItem`. I'm not 100% sure that it is better to go through `queryAttributes` here, because you only need to deal with outer attributes.